File: 392. Is Subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        k = 0
        for i in range(len(s)):
            a = t[k:].find(s[i])
            if a != -1:
                k += a+1
            else:
                return(False)
        return(True)

File: 392_IsSubsequence.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        int s_p = 0, t_p = 0;
        while (t_p < t.length()) {
            // end of string s -- all letters consecutively found
            if (s_p == s.length()) {
                return true;
            }

            // letter occurrence found -- now search for the next letter
            if (s[s_p] == t[t_p]) {
                ++s_p;
            }
            ++t_p;
        }

        if (s_p == s.length()) {
            return true;
        }
        return false;
    }
};

// Runtime: O(n)
// Memory: O(1)
